docs/guide/interactions/slash_commands.rst

```diff
@@ -0,0 +1,1548 @@
+what the bot defines in code during invocation.
+
+.. _parameters:
+
+Parameters
+-----------
+
+Since slash commands are defined by making Python functions, parameters are similarly defined with function parameters.
+
+Each parameter must have an assiociated type. This restricts what type of value a user can and cannot input.
+Types are specified in code through :pep:`526` function annotations.
+
+For example, the following command has a ``liquid`` string parameter:
+
+.. code-block:: python
+
+    @client.tree.command()
+    async def bottles(interaction: discord.Interaction, liquid: str):
+        await interaction.response.send_message(f'99 bottles of {liquid} on the wall!')
+
+On the client, parameters show up as these little black boxes that need to be filled out during invocation:
+
+.. image:: /images/guide/app_commands/bottles_command_preview.png
+    :width: 300
+
+Since this is a string parameter, virtually anything can be inputted (up to Discord's limits).
+
+Other parameter types are more restrictive - for example, if an integer parameter is added:
+
+.. code-block:: python
+
+    @client.tree.command()
+    async def bottles(interaction: discord.Interaction, liquid: str, amount: int):
+        await interaction.response.send_message(f'{amount} bottles of {liquid} on the wall!')
+
+Trying to enter a non-numeric character for ``amount`` will result with this red message:
+
+.. image:: /images/guide/app_commands/input_a_valid_integer.png
+    :width: 300
+
+Additionally, since both of these parameters are required, trying to skip one will result in this message in red:
+
+.. image:: /images/guide/app_commands/this_option_is_required.png
+    :width: 300
+
+Some parameter types have different modes of input.
+
+For example, annotating to :class:`~discord.User` will show a selection of users to
+pick from in the current context and :class:`~discord.Attachment` will show a file-dropbox.
+
+A full overview of supported types can be seen in the :ref:`type conversion table <type_conversion>`.
+
+typing.Optional
+++++++++++++++++
+
+Discord supports optional parameters, wherein a user doesn't need to provide a value during invocation.
+
+A parameter is considered optional if its assigned a default value and/or annotated
+to :obj:`~typing.Optional`.
+
+For example, this command displays a given user's avatar, or the current user's avatar:
+
+.. code-block:: python
+
+    from typing import Optional
+
+    @client.tree.command()
+    async def avatar(interaction: discord.Interaction, user: Optional[discord.User] = None):
+        avatar = (user or interaction.user).display_avatar
+        await interaction.response.send_message(avatar.url)
+
+On Discord:
+
+.. image:: /images/guide/app_commands/avatar_command_optional_preview.png
+    :width: 300
+
+`Python version 3.10+ union types <https://peps.python.org/pep-0604/>`_ are also supported instead of :obj:`typing.Optional`.
```

Comment:
```suggestion
:pep:`Python version 3.10+ union types <604>`_ are also supported instead of :obj:`typing.Optional`.
```
Again I wouldn't link to a PEP but this is slightly cleaner syntax, https://docs.python.org/3/library/stdtypes.html#types-union might be a better link.